### PR TITLE
Dice game

### PR DESCRIPTION
**Pull request by VolodymyrButs**

### CODE REVIEW

File: src/pages/Dice.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    const [historyData, setHistoryData] = useState([])
+    const [result, setResult] = useState(null)
+
+    const token = useToken()
+
+    const platform = window.navigator.userAgentData.mobile
+        ? 'Web-mobile'
+        : 'Web-desktop'
+
+    const getHistory = () => {
```
Comment (dkupchyk) [inline src/pages/Dice.js]: Suggest moving to a separate service file with all fetches.

File: src/pages/Dice.js [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    const [bet, setBet] = useState(null)
+    const [isLoading, setLoading] = useState(false)
+    const [showHistory, setShowHistory] = useState(false)
+    const [historyData, setHistoryData] = useState([])
+    const [result, setResult] = useState(null)
+
+    const token = useToken()
+
+    const platform = window.navigator.userAgentData.mobile
+        ? 'Web-mobile'
```
Comment (dkupchyk) [inline src/pages/Dice.js]: would move this variables to constants file

File: src/pages/Dice.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+            }, 1000)
+            return
+        }
+
+        const formData = {
+            dateTime: moment().format('DD-MM-YYYY hh:mm:ss'),
+            betType: bet,
+        }
+
+        fetch(
```
Comment (dkupchyk) [inline src/pages/Dice.js]: the same with service file

File: src/App.js [hunk 4]
```diff
@@ -33,6 +34,9 @@ function App() {
                 <Switch>
                     <Route path="/" exact component={MainPage} />
                     <Route path="/game/:id" component={GamePage} />
+                    <Route path="/dice">
+                        <Dice />
+                    </Route>
```
Comment (STDio96) [inline src/App.js]: why not use it the same way as it is above: `<Route path=... component={Dice} />`?

File: src/components/dice-game/Dice.js [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+            case 1:
+                return 2
+            case 2:
+                return 3
+            case 3:
+                return 4
+            case 4:
+                return 4
+            case 5:
+                return 6
```
Comment (STDio96) [inline src/components/dice-game/Dice.js:135]:
```
    case 0:
    case 1:
    case 2:
    case 3:
    case 4:
    case 5:
        return betIndex + 1;
```

**Review by dkupchyk — APPROVED** [review]

File: src/components/dice-game/Dice.js [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+                                ? `${styles.img} ${styles.isLoading}`
+                                : styles.img
+                        }
+                        src={DiceImg}
+                        title="dice"
+                    />
+                )}
+                {result !== null && (
+                    <h1 className={styles.result}>
+                        {isBetWin() ? 'ПОБЕДА !!!' : 'ПОРАЖЕНИЕ'}
```
Comment (STDio96) [inline src/components/dice-game/Dice.js:286]: move to constants

File: src/components/dice-game/Dice.js [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+                                            {getBetValue(historyItem.betType)} -
+                                            Результат:{' '}
+                                            {historyItem.droppedNumber} ПОБЕДА:{' '}
+                                            {historyItem.isBetWon ? '✅' : '❌'}
+                                        </p>
+                                        <hr />
+                                    </li>
+                                )
+                            })}
+                        </ol>{' '}
```
Comment (STDio96) [inline src/components/dice-game/Dice.js:330]: `{' '}` why?

File: src/components/dice-game/Dice.js [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+                                    <li
+                                        key={historyItem.id}
+                                        className={styles.historyListItem}
+                                    >
+                                        <p> Дата: {historyItem.dateTime}</p>
+                                        <p>
+                                            Ставка:{' '}
+                                            {getBetValue(historyItem.betType)} -
+                                            Результат:{' '}
+                                            {historyItem.droppedNumber} ПОБЕДА:{' '}
```
Comment (STDio96) [inline src/components/dice-game/Dice.js:323]: to constants later

**Review by STDio96 — COMMENTED** [review]
have a look, please

File: src/pages/Dice.js (hunk 1 above)
Comment (VolodymyrButs) [reply src/pages/Dice.js]: After demo

File: src/components/dice-game/Dice.js (hunk 5 above)
Comment (VolodymyrButs) [reply src/components/dice-game/Dice.js:135]: not obvious

File: src/components/dice-game/Dice.js (hunk 6 above)
Comment (VolodymyrButs) [reply src/components/dice-game/Dice.js:286]: use just in this case

File: src/components/dice-game/Dice.js (hunk 7 above)
Comment (VolodymyrButs) [reply src/components/dice-game/Dice.js:330]: gap
after the demo restyle the whole component

**Pull request merged**